### ISSUE TITLE
Language model for transducer model

### ISSUE THREAD

**Issue by DatNgoBK** [NVIDIA/NeMo#3412]
Can I use decoder with external language model for Transducer model ?

**Comment by VahidooX** [comment]
Currently our Transducer models support beam search decoding but not external LM. But you may use neural rescoring to rerank the candidates from the beam search decoder. The only issue is that our beam search evaluation script (https://github.com/NVIDIA/NeMo/blob/main/scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram.py) does not support Transducer models now. It should be easy to add this feature. Hopefully I would add it soon.

**Comment by Omarnabk** [comment]
@VahidooX is the Transducer able to return the logits of the transcript?

**Comment by titu1994** [comment]
No. RNNT cannot return logits because the decoding step is dynamic and Autoregressive unlike CTC - it returns a dangling tensor of shape (B, T*, U*, V) where * denotes dynamic computation so there's no easy way to have a filled logits tensor.

**Comment by VahidooX** [comment]
I created a draft PR here which adds the support of LM for Transducer models: 

https://github.com/NVIDIA/NeMo/pull/3760

You may use the following script to perform beam search decoding with or without N-gram LM with Transducer models. It also enables you to perform neural rescoring with Transducer models.

`scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram_transducer.py`

Please note that this PR is still in experimental stage and may need more updates in the API and more experiments. unfortunately, adding this feature along with documentation the main branch is on hold for now because of some other tasks with higher priority and it may take need more time.
